UnityProject/Assets/Scripts/Messages/Server/UpdateTileMessage.cs

```diff
@@ -10,14 +11,64 @@ public class UpdateTileMessage : ServerMessage
 	public Color Colour;
 	public uint TileChangeManager;
 
+	public static List<delayedData> DelayedStuff = new List<delayedData>();
+
+	public struct delayedData
+	{
+		public Vector3Int Position;
+		public TileType TileType;
+		public string TileName;
+		public Matrix4x4 TransformMatrix;
+		public Color Colour;
+		public uint TileChangeManager;
+
+		public delayedData(Vector3Int inPosition, TileType inTileType, string inTileName, Matrix4x4 inTransformMatrix,
+			Color inColour, uint inTileChangeManager)
+		{
+			Position = inPosition;
+			TileType = inTileType;
+			TileName = inTileName;
+			TransformMatrix = inTransformMatrix;
+			Colour = inColour;
+			TileChangeManager = inTileChangeManager;
+		}
+	}
+
+
 	public override void Process()
 	{
 		LoadNetworkObject(TileChangeManager);
-		var tileChangerManager = NetworkObject.GetComponent<TileChangeManager>();
-		tileChangerManager.InternalUpdateTile(Position, TileType, TileName, TransformMatrix, Colour);
+		if (NetworkObject == null)
+		{
+			DelayedStuff.Add(new delayedData(Position, TileType, TileName, TransformMatrix, Colour, TileChangeManager));
+		}
+		else
+		{
+			var tileChangerManager = NetworkObject.GetComponent<TileChangeManager>();
+			tileChangerManager.InternalUpdateTile(Position, TileType, TileName, TransformMatrix, Colour);
+			TryDoUndone();
+		}
+	}
+
+	public void TryDoUndone()
+	{
+		for (int i = 0; i < DelayedStuff.Count; i++)
+		{
+			NetworkObject = null;
+			LoadNetworkObject(DelayedStuff[i].TileChangeManager);
+			if (NetworkObject != null)
+			{
+				var tileChangerManager = NetworkObject.GetComponent<TileChangeManager>();
+				tileChangerManager.InternalUpdateTile(DelayedStuff[i].Position, DelayedStuff[i].TileType,
+					DelayedStuff[i].TileName, DelayedStuff[i].TransformMatrix, DelayedStuff[i].Colour);
+				DelayedStuff.RemoveAt(i);
+				i--;
```

Comment:
Codacy might be right, your loop is increasing but you're decreasing inside it. Is it intended?

Reply:
well I increment it down after removing element so,  it still in the same relative place
so If I remove zero it will still be at zero in the next pass

Reply:
Gotcha